### PR TITLE
Js.alloc

### PR DESCRIPTION
**Pull request by jskelcy**

### CODE REVIEW

**Comment by jskelcy** [comment]
Seemed like when users trying to run the vanilla paralyze function we are doing lots of unneeded allocs so we can use ParalyzeWithCancel.

In an attempt to speed things up for people who are not taking advantage of canceling requests I pulled out that logic. Here are the results of the benchmark (BenchmarkWithoutAlloc is using the new code) :

```
$ go test -bench=.  @-benchmem
PASS

BenchmarkWithChannelAlloc-8       200000          7031 ns/op        1600 B/op         28 allocs/op
BenchmarkWithoutChannelAlloc-8   1000000          2167 ns/op         256 B/op          9 allocs/op
```

As you can see there was a 5X speedup. Alas the lib now just wraps wait group with a little sugar to so that you have an arrays of results and errors, but the speedup with worth it.

**Comment by eculver** [comment]
BUMP, lgtm

**Comment by jskelcy** [comment]
Quick Note:

I had to change it so that results are _not_ mutually exclusive. If there are cases where both an error and result is returned both should be available to the main program for handling.

A quick example is in the case of a http redirect where there will be both a result and a error and I think in that case it is a much better pattern to return both return values for handling.

File: paralyze.go [hunk 1]
```diff
@@ -25,10 +25,24 @@ var (
...
+	results = make([]interface{}, len(funcs))
+	errors = make([]error, len(funcs))
+	wg.Add(len(funcs))
+
+	for i, fn := range funcs {
+		go func(i int, fn Paralyzable) {
+			defer wg.Done()
+			res, err := fn()
+			results[i] = res
+			errors[i] = err
```
Comment (i) [inline paralyze.go]: can just be

results[i], errors[i] = fn()

File: bench_test.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+import "testing"
+
+var (
+	fasterFn = func() (interface{}, error) { return 55, nil }
+)
+
+func BenchmarkWithChannelAlloc(test *testing.B) {
+	for i := 0; i < test.N; i++ {
+		ParalyzeWithTimeout(0, []Paralyzable{
```
Comment (i) [inline bench_test.go]: nit, this doesn't need to be a slice.

Comment (jskelcy) [inline bench_test.go]: yeah no idea why I did that.

File: paralyze.go (hunk 1 above)
Comment (jskelcy) [inline paralyze.go]: truth

**Comment by jskelcy** [comment]
@i updated

**Comment by i** [comment]
LGTM

**Pull request merged**